backoff.go

```diff
@@ -0,0 +1,11 @@
+package coap
+
+import (
+	"github.com/cenkalti/backoff/v4"
+)
+
+// Stop indicates that no more retries should be made for use in NextBackOff().
+const Stop = backoff.Stop
+
+// BackOff is a backoff policy for retrying an operation.
+type BackOff = backoff.BackOff
```

Comment:
I would prefer for the policy to be represented using the native types. The last time, we have agreed to wrap the policy in this interface:
```go
type RetryFunc = func() (when time.Time, err error)
type RetryFuncFactory = func() RetryFunc

func handleRetry(ctx context.Context, retryFunc RetryFunc) error {
	when, err := retryFunc()
	if err != nil {
		return fmt.Errorf("cannot retry: %v", err)
	}
	select {
	case <-time.After(time.Until(when)):
	case <-ctx.Done():
		return fmt.Errorf("retry canceled")
	}
	return nil
}
```